### PR TITLE
Decode connection url

### PR DESCRIPTION
**Pull request by cressie176**
See https://github.com/onebeyond/rascal/issues/227

### CODE REVIEW

File: lib/config/configure.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ module.exports = _.curry((rascalConfig, next) => {
     } = new URL(connectionString);
     const options = Array.from(searchParams).reduce((attributes, entry) => ({ ...attributes, [entry[0]]: entry[1] }), {});
     return {
-      protocol, hostname, port, user, password, vhost, options,
+      protocol, hostname: decodeURIComponent(hostname), port, user: decodeURIComponent(user), password: decodeURIComponent(password), vhost: decodeURIComponent(vhost), options,
```
Comment (toby-farris-bd) [inline lib/config/configure.js:94]: There is reason I wanted to have a config for the consumer to indicate what was already encoded in https://github.com/onebeyond/rascal/pull/228. What if there is a sequence in the `user` like `%23` that was not a product of encoding, but just part of an actual password? This code here will destructively decode it.

Comment (toby-farris-bd) [reply lib/config/configure.js:94]: Ha! I see you're testing that [below](https://github.com/onebeyond/rascal/pull/229/files#diff-59cc4a3af1181865c858de26aa6d223f2c332351d061d6e6025d24fcd190bfb5R83). Have to think about it a bit more to see if there is a case of a destructive decoding...

Comment (toby-farris-bd) [reply lib/config/configure.js:94]: > Ha! I see you're testing that [below](https://github.com/onebeyond/rascal/pull/229/files#diff-59cc4a3af1181865c858de26aa6d223f2c332351d061d6e6025d24fcd190bfb5R83). Have to think about it a bit more to see if there is a case of a destructive decoding...

No, I'm confusing myself. I think my original idea in the above PR is correct.

There is no way for rascal to know whether the consumer intended `#` as the character, or the set of `%23` to be in the password (or the other parts of the url).

What your test assumes is that a consumer that passes a url like

```
http://friend:%23Mellon@Khazad-dum.west.gate
```

doesn't actually want rascal to encode it to 

```
http://friend:%2523Mellon@Khazad-dum.west.gate
``` 

Does that make sense?

Comment (toby-farris-bd) [reply lib/config/configure.js:94]: I would also argue this is a breaking change

File: lib/config/configure.js [hunk 2]
```diff
@@ -105,10 +105,14 @@ module.exports = _.curry((rascalConfig, next) => {
...
-    connection.management.url = connection.management.url || url.format({ ...connection.management, auth });
-    connection.management.loggableUrl = connection.management.url.replace(/:[^:]*?@/, ':***@');
+    connection.management = _.isString(connection.management) ? { url: connection.management } : connection.management;
+    const attributesFromUrl = parseConnectionUrl(connection.management.url);
+    const attributesFromConfig = getConnectionAttributes(connection.management);
+    const defaults = { user: connection.user, password: connection.password, hostname: connection.hostname };
+
+    const connectionAttributes = _.defaultsDeep({ options: null }, attributesFromUrl, attributesFromConfig, defaults);
+    setConnectionAttributes(connection.management, connectionAttributes);
+    setConnectionUrls(connection.management);
```
Comment (cressie176) [inline lib/config/configure.js:115]: We needed to do something similar with the connection management url too.

**Pull request merged**